pvlib/singlediode.py

```diff
@@ -150,21 +173,49 @@ def bishop88(diode_voltage, photocurrent, saturation_current,
     # calculate temporary values to simplify calculations
     v_star = diode_voltage / nNsVth  # non-dimensional diode voltage
     g_sh = 1.0 / resistance_shunt  # conductance
-    i = (photocurrent - saturation_current * np.expm1(v_star)
-         - diode_voltage * g_sh - i_recomb)
+    if breakdown_factor > 0:  # reverse bias is considered
+        brk_term = 1 - diode_voltage / breakdown_voltage
+        brk_pwr = np.power(brk_term, -breakdown_exp)
+        i_breakdown = breakdown_factor * diode_voltage * g_sh * brk_pwr
+    else:
+        i_breakdown = 0.
+    i = (
+        photocurrent
+        - saturation_current * np.expm1(v_star)
+        - diode_voltage * g_sh
+        - i_recomb
+    )
+         diode_voltage * g_sh - i_recomb - i_breakdown)
     v = diode_voltage - i * resistance_series
     retval = (i, v, i*v)
     if gradients:
         # calculate recombination loss current gradients where d2mutau > 0
         grad_i_recomb = np.where(is_recomb, i_recomb / v_recomb, 0)
         grad_2i_recomb = np.where(is_recomb, 2 * grad_i_recomb / v_recomb, 0)
         g_diode = saturation_current * np.exp(v_star) / nNsVth  # conductance
-        grad_i = -g_diode - g_sh - grad_i_recomb  # di/dvd
+        if breakdown_factor > 0:  # reverse bias is considered
+            brk_pwr_1 = np.power(brk_term, -breakdown_exp - 1)
+            brk_pwr_2 = np.power(brk_term, -breakdown_exp - 2)
+            brk_fctr = breakdown_factor * g_sh
+            grad_i_brk = brk_fctr * (brk_pwr + diode_voltage *
+                                     -breakdown_exp * brk_pwr_1)
+            grad2i_brk = brk_fctr * -breakdown_exp * (
+            grad2i_brk = (
+                brk_fctr
+                * -breakdown_exp
+                * (2 * brk_pwr_1 + diode_voltage * (-breakdown_exp - 1) * brk_pwr_2)
+            )
```

Comment:
I don't fully understand the stickler comments here. 

`-breakdown_exp` is used a lot. I recommend new variable for this. I also don't know the python operation binding rules well enough to immediately see that the math is going to be evaluated as intended.